### PR TITLE
[EBPF] Add more recognized error types to KMT

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

Adds two more recognized error types to KMT errors.

### Motivation

Reduce failure rates, have more visibility into failures by emitting metrics.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

File: test/new-e2e/system-probe/errors.go [hunk 1]
```diff
@@ -104,6 +106,18 @@ var handledErrorsLs = []handledError{
...
+		errorType:   ec2InstanceCreateTimeout,
+		errorString: "creating EC2 Instance: operation error",
+		metric:      "ec2-instance-create-timeout",
+		action:      retryStack | emitMetric,
+	},
+	{
+		errorType:   ddAgentRepoFailure,
+		errorString: "Failed to update the sources after adding the Datadog repository.",
+		metric:      "apt-dd-agent-repo-failure",
+		action:      retryStack | emitMetric,
```
Comment (usamasaqib) [inline test/new-e2e/system-probe/errors.go:119]: Do we know if we can gracefully retry from this point? If not, maybe we should just emit a metric right now until retrying is supported.

Comment (gjulianm) [reply test/new-e2e/system-probe/errors.go:119]: I think we can. I found this error message in [this job](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/636589328#L2500), it was a problem where we ran a job just as the repositories were being updated, and the built-in retry mechanism for [the command](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/636589328#L2308) wasn't enough for it pass. I didn't notice that retries were built-in, but maybe retrying from the stack gives enough time for these rare cases to pass without breaking the test.

Comment (gjulianm) [reply test/new-e2e/system-probe/errors.go:119]: Will remove the retry in that case until we look further into it

**Pull request merged**